yarn-project/simulator/src/public/enqueued_calls_processor.ts

```diff
@@ -265,17 +331,28 @@ export class EnqueuedCallsProcessor {
         );
         throw enqueuedCallResult.revertReason;
       }
+      await txStateManager.mergeStateForEnqueuedCall(
+        enqueuedCallStateManager,
+        callRequest,
+        executionRequest.args,
+        enqueuedCallResult.reverted!,
+      );
 
       avmProvingRequest = enqueuedCallResult.avmProvingRequest;
       returnValues.push(enqueuedCallResult.returnValues);
       gasUsed = gasUsed.add(enqueuedCallResult.gasUsed);
+      reverted = enqueuedCallResult.reverted;
       revertReason ??= enqueuedCallResult.revertReason;
 
+      // Instead of operating on worldStateDB here, do we do AvmPersistableStateManager.revert() or return()?
       if (revertReason) {
         // TODO(#6464): Should we allow emitting contracts in the private setup phase?
         // if so, this is removing contracts deployed in private setup
+        // You can't submit contracts in public, so this is only relevant for private-created
+        // side effects
+        // Are we reverting here back to end of non-revertible insertions?
+        // What are we reverting back to?
```

Comment:
Yes, we are reverting back to end of non-revertible. If a contract is deployed in private in revertible phase (it's most likely be the case), it should be removed here. So that it can't be called in teardown.